src/NLog/LayoutRenderers/AllEventPropertiesLayoutRenderer.cs

```diff
@@ -80,18 +84,50 @@ public AllEventPropertiesLayoutRenderer()
         [DefaultValue(false)]
         public bool IncludeEmptyValues { get; set; } = false;
 
-#if NET4_5
+        /// <summary>
+        /// Gets or sets the keys to exclude from the output. If omitted, none are excluded.
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+#if NET3_5
+        public HashSet<string> Exclude { get; set; }
+#else
+        public ISet<string> Exclude { get; set; }
+#endif
 
+#if NET4_5
         /// <summary>
         /// Also render the caller information attributes? (<see cref="System.Runtime.CompilerServices.CallerMemberNameAttribute"/>,
         /// <see cref="System.Runtime.CompilerServices.CallerFilePathAttribute"/>, <see cref="System.Runtime.CompilerServices.CallerLineNumberAttribute"/>). 
         /// 
         /// See https://msdn.microsoft.com/en-us/library/hh534540.aspx
         /// </summary>
         /// <docgen category='Rendering Options' order='10' />
+        [Obsolete("Instead use the Exclude-property. Marked obsolete on NLog 4.6.8")]
         [DefaultValue(false)]
-        public bool IncludeCallerInformation { get; set; }
-
+        public bool IncludeCallerInformation
+        {
+            get => Exclude?.Contains(CallerInformationAttributeNames[0]) != true;
+            set
```

Comment:
I think it's better to keep the separate boolean property/field and something like `!Exclude.Contains(..) && !IsUnwantedCallerPropery(..)`   . That makes things a lot easier and also better to remove in the future

Reply:
I want to remove IncludeCallerInformation and remove CallerAttributes-properties in NLog 5. See no reason to introduce new stuff to remove right after.

Reply:
IMO this is introducing new stuff, but I see why it's introduced so let's keep it like this.